onnxruntime/python/dlpack/dlpack_converter.cc

```diff
@@ -175,7 +175,7 @@ const char* GetOrtDeviceName(const OrtDevice& device) {
 }
 
 bool IsContiguousTensor(const DLTensor& tensor) {
-  if (!tensor.strides) {
+  if (!tensor.strides || (tensor.ndim == 1 && tensor.shape[0] == 1 && tensor.strides[0] == 0)) {
```

Comment:
@thiagocrepaldi , I actually found how PT doing this here: https://github.com/pytorch/pytorch/blob/33f206b865ac1fa72c3f5a4ebef5e160cfb12f6c/aten/src/ATen/TensorUtils.cpp#L267, so I think we can change it like this:

`bool IsContiguousTensor(const DLTensor& tensor) {
  if (!tensor.strides) {
    return true;
  }

  int64_t running_size = 1;
  for (int i = tensor.ndim - 1; i >= 0; i--) {
    if (tensor.shape[i] == 0) {
      return true;
    }

    if (tensor.shape[i] != 1 && tensor.strides[i] != running_size) {
      return false;
    }

    running_size *= tensor.shape[i];
  }

  return true;
}`

Reply:
nice!